channels/rail/client/rail_orders.c

```diff
@@ -536,9 +546,17 @@ static UINT rail_recv_handshake_ex_order(railPlugin* rail, RAIL_HANDSHAKE_EX_ORD
 		return error;
 	}
 
+	clientHandshake.buildNumber = 0x00001DB0;
+	/* 2.2.2.2.3 HandshakeEx PDU (TS_RAIL_ORDER_HANDSHAKE_EX)
+	 * Client response is really a Handshake PDU */
+	error = context->ClientHandshake(context, &clientHandshake);
+
+	if (error != CHANNEL_RC_OK)
+		return error;
+
 	if (context->custom)
 	{
-		IFCALLRET(context->ClientHandshakeEx, error, context, handshakeEx);
+		IFCALLRET(context->ServerHandshakeEx, error, context, handshakeEx);
```

Comment:
Going through this PR for an implementation of rail server support I'm working on, noticed that you implemented here a change that got merged in #5345